lib/bootloader.js

```diff
@@ -45,7 +45,12 @@
   // version without applicationCache intercepting us.
   function getShadowIndex(success, failure) {
     var req = new XMLHttpRequest();
-    req.open('GET', 'APPSHELL_PATHappshell.html', true);
+    var path = 'APPSHELL_PATHappshell';
+    var query = window.location.search.match(/revision=([^&#]+)/);
```

Comment:
The same behavior, revision is stored in appcache using key including revision query string. Offline the app will attempt to load from the appcache if revision in query string is the same otherwise it won't.